### PR TITLE
MKB-658: create Branch content type

### PR DESCRIPTION
**Pull request by radoslavzeman**
- create one common content type for all Localities / Premises / Event Localities / Places / Departments...
- `eventLocality` relation in Events migrated to `branch` relation

The idea is, that admin will chose which Branches to show where (on homepage, next to map on Navstivte nas page, in lint of places to rent...)

- remove `Event Locality` content type
- change event filters to accept branch instead of eventLocality
- add "services" to Branch detail page

Note: Localities in menu are still Pages. They have the same slug as branches so next config rewrites it to branch detial page. - TODO

TODO:
- create page section for places to rent
- remove `premises` layout
- remove `Premise` content type
- add events to Branch detail page
- add possibility to chose from Branches in Menu
- finally, remove old locality pages

### CODE REVIEW

File: next/utils/getBranchInfo.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { BranchPlaceNestedEntityFragment } from '@bratislava/strapi-sdk-city-library'
+
+export const getBranchInfo = (branch: BranchPlaceNestedEntityFragment | null | undefined) => {
```
Comment (MarekBodingerBA) [inline next/utils/getBranchInfo.ts:3]: It's clear what code does, but the "business" reason is not clear.

What is `parentBranch`, why is the address used from there if present? Please add comment.

File: next/graphql/queries/Events.graphql [hunk 2]
```diff
@@ -68,6 +61,31 @@ fragment EventCardAttributes on Event {
...
+  }
+}
+
+fragment BranchPlaceNestedEntity on BranchEntity {
+  ...BranchPlaceEntity
+  attributes {
+    parentBranch {
+      data {
+        id
+        attributes {
```
Comment (MarekBodingerBA) [inline next/graphql/queries/Events.graphql:79]: `...BranchPlaceEntity`?

File: strapi/src/components/blocks/opening-hours-item.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (MarekBodingerBA) [inline strapi/src/components/blocks/opening-hours-item.json:1]: Is this enough? We have a specific field for each day now.

**Review by MarekBodingerBA — APPROVED** [review]

File: strapi/src/components/blocks/opening-hours-item.json (hunk 3 above)
Comment (MarekBodingerBA) [reply strapi/src/components/blocks/opening-hours-item.json:1]: image

File: next/utils/getBranchInfo.ts (hunk 1 above)
Comment (radoslavzeman) [reply next/utils/getBranchInfo.ts:3]: Noted to TODOs

File: next/components/Molecules/LocalityDetails/LocalityDetailsServices.tsx [hunk 4]
```diff
@@ -1,39 +1,51 @@
...
-            uppercase={false}
-            customIcon={
-              <span className="ml-2 inline-flex">
-                <ChevronRightSvg />
-              </span>
-            }
-            key={service?.page?.data?.id ?? ''}
-          />
-        ))}
+        {branch.attributes.servicePages.data.filter(isDefined).map((service) => {
```
Comment (MarekBodingerBA) [inline next/components/Molecules/LocalityDetails/LocalityDetailsServices.tsx:23]: `.filter(isDefined).filter(branch => isDefined(branch.attributes))` ?

**Pull request merged**